internal/policy/container/has_modified_files.go

```diff
@@ -24,140 +28,158 @@ var _ check.Check = &HasModifiedFilesCheck{}
+					HasRPMDB:          false,
+				}
+				continue
+			}
 
-// dropFirstLayerIfEmpty will evaluate the length of the first layer and will drop its entry in files if it's empty.
-// This avoids establishing a baseline of files installed in the original layer with an empty list. An example
-// of when this should occur is in cases where the image was built FROM scratch, such as ubi-micro. This is required
-// to ensure we don't attempt to validate against an empty layer.
-func (p *HasModifiedFilesCheck) dropFirstLayerIfEmpty(files [][]string) ([][]string, bool) {
-	var dropped bool
-	if len(files[0]) == 0 {
-		files = files[1:] // shift the empty layer out.
-		dropped = true
+			// If it's the first layer, just make the pkgList empty.
+			pkgList = make([]*rpmdb.PackageInfo, 0)
+		}
+
+		pkgNameList := extractPackageNameVersionRelease(pkgList)
+
+		packageFiles, err := installedFileMapWithExclusions(ctx, pkgList)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		layerRefs[layerID] = packageFilesRef{
+			LayerFiles:        files,
+			LayerPackages:     pkgNameList,
+			LayerPackageFiles: packageFiles,
+			HasRPMDB:          true,
+		}
 	}
 
-	return files, dropped
+	return layerIDs, layerRefs, nil
 }
 
 // validate compares the list of LayerFiles and PackageFiles to see what PackageFiles
 // have been modified within the additional layers.
-//
-//nolint:unparam // ctx is unused. Keep for future use.
-func (p *HasModifiedFilesCheck) validate(ctx context.Context, packageFilesRef *packageFilesRef) (bool, error) {
+func (p *HasModifiedFilesCheck) validate(ctx context.Context, layerIDs []string, packageFiles map[string]packageFilesRef) (bool, error) {
 	logger := logr.FromContextOrDiscard(ctx)
 
-	layerFiles := packageFilesRef.LayerFiles
-	packageFiles := packageFilesRef.PackageFiles
-
-	// Determine the list of files that were a part of the base layer.
-	baseLayer := make(map[string]struct{}, len(layerFiles[0]))
-	for _, filename := range layerFiles[0] {
-		if _, ok := packageFiles[filename]; ok {
-			baseLayer[filename] = struct{}{}
-		}
-	}
-	layers := layerFiles[1:]
-
-	// Look for modifications in subsequent layers by determining
-	// if a file in a base layer exists in a subsequent layer.
-	modifiedFilesDetected := false
-	for _, layer := range layers {
-		for _, file := range layer {
-			if _, ok := baseLayer[file]; ok {
-				// This means the files exists in the base layer. This is a fail.
-				logger.V(log.DBG).Info("modified file detected", "filename", file)
-				modifiedFilesDetected = true
+	disallowedModifications := map[string]string{}
+	for idx, layerID := range layerIDs {
+		ref := packageFiles[layerID]
+		// if ref.HasRPMDB {
+		// 	// TODO: bpc
+		// 	// Something else needs to be other than just skipping because we have an RPM db in this layer.
+		// 	// It works for the *first* layer with an RPM db, since everything would be flagged, but we need to
+		// 	// figure out WHY a new RPM db is present.
+		// 	continue
+		// }
```

Comment:
Is this still valid to be commented out? I think you covered this in other logic where you are finding the RPM's.

Reply:
Oh. Meant to remove that.